wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java

```diff
@@ -164,6 +202,89 @@ private String getTargetHPCCBuildVersionString() throws Exception
 
     }
 
+    public SpanBuilder getSpanBuilder(String spanName)
+    {
+        Tracer tracer = getTracer(INSTRUMENTED_LIBRARY_NAME);
+        SpanBuilder spanBuilder = tracer.spanBuilder(spanName)
+          .setAttribute(ServerAttributes.SERVER_ADDRESS, wsconn.getHost())
+          .setAttribute(ServerAttributes.SERVER_PORT, Long.getLong(wsconn.getPort()))
+          .setAttribute(HttpAttributes.HTTP_REQUEST_METHOD, HttpAttributes.HttpRequestMethodValues.GET)
+          .setSpanKind(SpanKind.CLIENT);
+        //.startSpan();
+
+        return spanBuilder;
+    }
+
+    public void injectTraceParentHeader(Span currentSpan)
+    {
+        injectTraceParentHeader(stub._getServiceClient().getOptions(), currentSpan);
+    }
+
+    static public void injectTraceParentHeader(Options options, Span currentSpan)
+    {
+        if (options != null)
+        {
+            if (currentSpan != null && currentSpan.getSpanContext().isValid())
+                W3CTraceContextPropagator.getInstance().inject(Context.current(), options, Options::setProperty);
```

Comment:
should we propagate the Context.current(), or the currentSpan.getContext() passed in?

Reply:
I think we should use currentSpan.getContext() for consistency. We could also have an override of injectTraceParentHeader() that doesn't require any input and grabs the current span via Span.current()